### PR TITLE
Fixes for index.ts, string methods/array.length

### PR DESCRIPTION
**Pull request by grubsteak**

### CODE REVIEW

File: src/compiler/call.ts [hunk 1]
```diff
@@ -479,11 +481,11 @@ export function getPropertyAccessExpressionType(
 	const subExpType = subExp.getType();
 	const property = expression.getName();
 
-	if (isArrayMethodType(expType)) {
+	if (isArrayMethodType(expType) || (isArrayType(subExpType) && property === "length")) {
```
Comment (grubsteak) [inline src/compiler/call.ts]: this probably isn't the best way to do this

**Comment by evaera** [comment]
Please provide a more full description of the changes present in this PR and the motivations behind them

**Comment by Validark** [comment]
This is not a proper fix for the .length issue. Please remove that. My PR https://github.com/roblox-ts/roblox-ts/pull/340 will fix that.

File: src/compiler/call.ts (hunk 1 above)
Comment (Validark) [reply src/compiler/call.ts]: It isn't. I already have a fix for this in my PR https://github.com/roblox-ts/roblox-ts/pull/340. Please remove this.

**Pull request closed without merging**